src/node/rpc/frontend.h

```diff
@@ -566,7 +577,9 @@ namespace ccf
         }
 
         // Client signature is only recorded on the primary
-        if (consensus == nullptr || consensus->is_primary())
+        if (
+          consensus == nullptr || consensus->is_primary() ||
```

Comment:
This check could be in `record_client_signature()`, i.e. `if (client_signatures == nullptr) return;`

Reply:
discussed offline